activemq-client/src/main/java/org/apache/activemq/command/ActiveMQMessage.java

```diff
@@ -571,6 +603,7 @@ else if (AMQ_SCHEDULED_CRON.equals(name)) {
 
     @Override
     public Object getObjectProperty(String name) throws JMSException {
+        checkIsMessageAccessibleAndMutable();
```

Comment:
I was trying to come up with a neater way of doing this check. I got a neat annotation to work using aspect-j. It looks a lot cleaner but overall I don't think it's worthwhile as it introduces a extra dependency to the pom.

I'll continue pondering......